docs/ROOTFS_DISCOVERY.md

```diff
@@ -0,0 +1,407 @@
+---
+title: Boot Components & Root File System Discovery
+category: Booting
+layout: default
+SPDX-License-Identifier: LGPL-2.1-or-later
+---
+
+# Boot Components & Root File System Discovery
+
+The recommended way to boot a [`systemd`](https://systemd.io/) based
+[UEFI](https://en.wikipedia.org/wiki/UEFI) system consists primarily of three
+components:
+
+1. A boot loader,
+   i.e. [`systemd-boot`](https://www.freedesktop.org/software/systemd/man/latest/systemd-boot.html)
+   that provides interactive and programmatic control of what precisely to
+   boot. It takes care of enumerating all possible boot targets (implementing
```

Comment:
"It enumerates possible boot targets". (Not "all", there are always more possible that sd-boot doesn't know about. And "takes care of" is just word fluff.)